lib/chef/win32/security.rb

```diff
@@ -341,6 +341,21 @@ def self.get_token_information_primary_group(token)
         SID.new(group_result[:PrimaryGroup], group_result_storage)
       end
 
+      def self.get_token_information_elevation_type(token)
+        token_result_size = FFI::MemoryPointer.new(:ulong)
+        if GetTokenInformation(token.handle.handle, :TokenElevationType, nil, 0, token_result_size)
+          raise "Expected ERROR_INSUFFICIENT_BUFFER from GetTokenInformation, and got no error!"
+        elsif FFI::LastError.error != ERROR_INSUFFICIENT_BUFFER
+          Chef::ReservedNames::Win32::Error.raise!
+        end
+        info_ptr = FFI::MemoryPointer.new(:pointer)
+        token_info_pointer = TOKEN_ELEVATION_TYPE.new info_ptr
+        unless GetTokenInformation(token.handle.handle, :TokenElevationType, token_info_pointer, 4, token_result_size)
```

Comment:
Is there a `sizeof` function we can use instead of 4?

Reply:
Ideally it should have been `size_of(token_info_pointer)`, but it's initially 0. So that gives error. I have introduced a variable `token_info_length` for better readability. Please check if it looks fine.